lifecycle-operator/controllers/lifecycle/keptnworkloadinstance/controller_test.go

```diff
@@ -807,6 +808,161 @@ func TestKeptnWorkloadInstanceReconciler_ReconcileReachCompletion(t *testing.T)
 	}
 }
 
+func TestKeptnWorkloadInstanceReconciler_ReconcileReachCompletion_schedullingGates(t *testing.T) {
+
+	testNamespace := "some-ns"
+
+	wi := &klcv1alpha3.KeptnWorkloadInstance{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "some-wi",
+			Namespace: testNamespace,
+		},
+		Spec: klcv1alpha3.KeptnWorkloadInstanceSpec{
+			KeptnWorkloadSpec: klcv1alpha3.KeptnWorkloadSpec{
+				AppName: "some-app",
+				Version: "1.0.0",
+			},
+			WorkloadName:    "some-app-some-workload",
+			PreviousVersion: "",
+			TraceId:         nil,
+		},
+		Status: klcv1alpha3.KeptnWorkloadInstanceStatus{
+			DeploymentStatus:               apicommon.StateSucceeded,
+			PreDeploymentStatus:            apicommon.StateSucceeded,
+			PostDeploymentStatus:           apicommon.StateSucceeded,
+			PreDeploymentEvaluationStatus:  apicommon.StateSucceeded,
+			PostDeploymentEvaluationStatus: apicommon.StateSucceeded,
+			CurrentPhase:                   apicommon.PhaseWorkloadPostEvaluation.ShortName,
+			Status:                         apicommon.StateSucceeded,
+			StartTime:                      metav1.Time{},
+			EndTime:                        metav1.Time{},
+		},
+	}
+
+	app := controllercommon.ReturnAppVersion(
+		testNamespace,
+		"some-app",
+		"1.0.0",
+		[]klcv1alpha3.KeptnWorkloadRef{
+			{
+				Name:    "some-workload",
+				Version: "1.0.0",
+			},
+		},
+		klcv1alpha3.KeptnAppVersionStatus{
+			PreDeploymentEvaluationStatus: apicommon.StateSucceeded,
+		},
+	)
+
+	ch := make(chan bool)
+	var called bool
+	go func() {
+		called = <-ch
+	}()
+	r, eventChannel, _ := setupReconciler(wi, app)
+	r.SchedullingGatesEnabled = true
+	r.RemoveGates = func(ctx context.Context, c client.Client, log logr.Logger, workloadInstance *klcv1alpha3.KeptnWorkloadInstance) error {
+		ch <- true
+		return nil
+	}
+
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      "some-wi",
+		},
+	}
+
+	result, err := r.Reconcile(context.TODO(), req)
+
+	require.True(t, called)
+	require.Nil(t, err)
+
+	// do not requeue since we reached completion
+	require.False(t, result.Requeue)
+
+	// here we do not expect an event about the application preEvaluation being finished since that  will have been sent in
+	// one of the previous reconciliation loops that lead to the first phase being reached
+	expectedEvents := []string{
+		"CompletedFinished",
+	}
+
+	for _, e := range expectedEvents {
+		event := <-eventChannel
```

Comment:
let's put this into a `select` block with a timeout case to prevent the unit test from pending indefinitely if we do not receive the expected events